pkg/apis/migration/v1alpha1/model.go

```diff
@@ -161,3 +167,81 @@ func GetSecret(client k8sclient.Client, ref *kapi.ObjectReference) (*kapi.Secret
 
 	return &object, err
 }
+
+// DeleteMigrated - dispatches delete requests for migrated resources
+func DeleteMigrated(config *rest.Config, uid string) error {
+	GVRs, err := getGVRs(config)
+	if err != nil {
+		return err
+	}
+	client, err := dynamic.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+	listOptions := k8sclient.MatchingLabels(map[string]string{
+		MigratedByLabel: uid,
+	}).AsListOptions()
+	foreground := metav1.DeletePropagationForeground
+	deleteOptions := (&k8sclient.DeleteOptions{
+		PropagationPolicy: &foreground,
+	}).AsDeleteOptions()
+
+	for _, gvr := range GVRs {
+		list, err := client.Resource(gvr).List(*listOptions)
+		if err != nil {
+			return err
+		}
+		for _, r := range list.Items {
+			err = client.Resource(gvr).Namespace(r.GetNamespace()).Delete(r.GetName(), deleteOptions)
+			if err != nil {
+				return err
+			}
+		}
+	}
```

Comment:
This whole block could be simplified by using `DeleteCollection` method, but I hit issues with `ConfigMaps` while doing that.

```bash
oc api-resources -o wide | grep configmap
configmaps                            cm                                                     true         ConfigMap                            [create delete deletecollection get list patch update watch]
```

`
{"level":"error","ts":1586180540.0871735,"logger":"migration|z7pc5","msg":"","migration":"eb86bc20-75a9-11ea-bada-edebfcfc413f","error":"Resource - /v1, Resource=configmaps: the server does not allow this method on the requested resource","stacktrace":"github.com/konveyor/mig-controller/vendor/github.com/go-logr/zapr.(*zapLogger).Error\n\t/home/dgrigore/go/src/github.com/konveyor/mig-controller/vendor/github.com/go-logr/zapr/zapr.go:128\ngithub.com/konveyor/mig-controller/pkg/logging.Logger.Error\n\t/home/dgrigore/go/src/github.com/konveyor/mig-controller/pkg/logging/logger.go:75\ngithub.com/konveyor/mig-controller/pkg/logging.Logger.Trace\n\t/home/dgrigore/go/src/github.com/konveyor/mig-controller/pkg/logging/logger.go:81\ngithub.com/konveyor/mig-controller/pkg/controller/migmigration.(*Task).deleteMigrated\n\t/home/dgrigore/go/src/github.com/konveyor/mig-controller/pkg/controller/migmigration/restore.go:301\ngithub.com/konveyor/mig-controller/pkg/controller/migmigration.(*Task).Run\n\t/home/dgrigore/go/src/github.com/konveyor/mig-controller/pkg/controller/migmigration/task.go:534\ngithub.com/konveyor/mig-controller/pkg/controller/migmigration.(*ReconcileMigMigration).migrate\n\t/home/dgrigore/go/src/github.com/konveyor/mig-controller/pkg/controller/migmigration/migrate.go:78\ngithub.com/konveyor/mig-controller/pkg/controller/migmigration.(*ReconcileMigMigration).Reconcile\n\t/home/dgrigore/go/src/github.com/konveyor/mig-controller/pkg/controller/migmigration/migmigration_controller.go:195\ngithub.com/konveyor/mig-controller/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/home/dgrigore/go/src/github.com/konveyor/mig-controller/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:215\ngithub.com/konveyor/mig-controller/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1\n\t/home/dgrigore/go/src/github.com/konveyor/mig-controller/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:158\ngithub.com/konveyor/mig-controller/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1\n\t/home/dgrigore/go/src/github.com/konveyor/mig-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133\ngithub.com/konveyor/mig-controller/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/home/dgrigore/go/src/github.com/konveyor/mig-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134\ngithub.com/konveyor/mig-controller/vendor/k8s.io/apimachinery/pkg/util/wait.Until\n\t/home/dgrigore/go/src/github.com/konveyor/mig-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88"}
`